### ISSUE TITLE
Bash 4.2 generates malformed meson-cross.mingw.txt

### ISSUE THREAD

**Issue by litzh** [rdp/ffmpeg-windows-build-helpers#473]
bash-4.3-alpha introduced [a breaking change](https://github.com/bminor/bash/blob/master/CHANGES#L2303):
> When using the pattern substitution word expansion, bash now runs the replacement string through quote removal, since it allows quotes in that string to act as escape characters.  This is not backwards compatible, so it can be disabled by setting the bash compatibility mode to 4.2.

The following code is used to replace whitespace by `,` in `FLAGS`
```bash
cat >> meson-cross.mingw.txt << EOF

[properties]
c_args = ['${CFLAGS// /','}']
c_link_args = ['${LDFLAGS// /','}']
cpp_args = ['${CXXFLAGS// /','}']
cpp_link_args = ['${LDFLAGS// /','}']
EOF
```

Below is the result in bash-4.2.
```ini
[properties]
c_args = ['-mtune=generic\',\'-O3\',\'-pthread']
c_link_args = ['-pthread']
cpp_args = ['-mtune=generic\',\'-O3\',\'-pthread\',\'-pthread']
cpp_link_args = ['-pthread']
```

Below is the expected results.
```ini
[properties]
c_args = ['-mtune=generic','-O3','-pthread']
c_link_args = ['-pthread']
cpp_args = ['-mtune=generic','-O3','-pthread','-pthread']
cpp_link_args = ['-pthread']
```

**Comment by rdp** [comment]
Any ideas @PeacefullyDisturbed ?

**Comment by litzh** [comment]
This works for me, in both bash 4.2 and 5.0.
```bash
CFLAGS="-mtune=generic -O3 -pthread"
LDFLAGS="-pthread"
CXXFLAGS="-mtune=generic -O3 -pthread -pthread"

cat >> meson-cross.mingw.txt << EOF

[properties]
c_args = ['$(echo ${CFLAGS}|sed "s/ /\',\'/g")']
c_link_args = ['$(echo ${LDFLAGS}|sed "s/ /\',\'/g")']
cpp_args = ['$(echo ${CXXFLAGS}|sed "s/ /\',\'/g")']
cpp_link_args = ['$(echo ${LDFLAGS}|sed "s/ /\',\'/g")']
EOF
```

**Comment by PeacefullyDisturbed** [comment]
That change works for me with Bash 4.4 and 5.0 as well, so I think a pull request would be a good idea.

**Comment by PeacefullyDisturbed** [comment]
I ran into another issue with this function while testing #480, so in the course of fixing that I also changed it to use the pattern you recommended for the substitution.  So this should be fixed once #481 is merged.
